clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -154,15 +154,18 @@ object UncommittedGarbageCollector {
         // Process uncommitted
         val uncommittedGCRunInfo =
           new APIUncommittedAddressLister(apiClient).listUncommittedAddresses(spark, repo)
+        val uncommittedURI = new URI(uncommittedGCRunInfo.uncommittedLocation)
+        val uri = new URI(new URI(storageNamespace).getScheme,
+                          uncommittedURI.getHost,
+                          uncommittedURI.getPath,
+                          null
+                         )
```

Comment:
My main concern is that working with azure will not support this URI construction - at least I am sure we will require to test it.
Would feel better if we can use existing code or the original address to verify exists.

Reply:
Fixed